### PR TITLE
Feature/datastore search v1.1

### PR DESCRIPTION
**Pull request by leomrocha**
Basic New API implementation with:
- graphql wrapper
- datasetore_search

Documentation, Analysis and Examples are also as Markdown files with example screenshots added to the *documentation /* subdirectory

### CODE REVIEW

**Comment by EvgeniiaVak** [comment]
@leomrocha right now the prettier check is failing:
https://hub.docker.com/repository/registry-1.docker.io/datopian/data-api/builds/c05c1271-ec2f-4dc5-8e58-75cbcecd648f

```
$ /usr/src/app/node_modules/.bin/prettier --check .
Checking formatting...
[91m[warn] ANALYSIS.md
[0m
[91m[warn] README.md
[0m
[91m[warn] Code style issues found in the above file(s). Forgot to run Prettier?
[0m
[91merror Command failed with exit code 1.
[0m
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Please make sure you use prettier. `prittier --write` should run by default when you save files (because of vscode settings, probably you'll need a plugin for vscode for this. webstorm has it pre-built, but need to switch on).

**Comment by EvgeniiaVak** [comment]
Related explanation of builds in dockerhub ci and pull request checks: https://github.com/datopian/data-api/pull/14  @leomrocha please review

File: ANALYSIS.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# CKAN - New API Differences and Analysis
+
+Here we document the differences between this new API and [CKAN]() version
+
+By default all parameters and results from the new API are JSON, this differs from CKAN in which the get call parameters are non standard but have an ad-hoc syntax. This is to
```
Comment (EvgeniiaVak) [inline ANALYSIS.md:5]: Is something missing here?

File: ANALYSIS.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+## General Differences CKAN vs New API
+
+The new API is entirely based on a GraphQL DB querying system, this changes the ways in which we can make queries.
+
+CKAN targets a Postgresql DB, the difference with a GraphQL DB is that the queries that can be implemented are quite different.
+
+GraphQL queries need to be explicit, including even the fields ("columns" in a traditional DB), which means knowing the database and _"table"_ schema before making the query to be able to ask for the needed fields. This means that for
```
Comment (EvgeniiaVak) [inline ANALYSIS.md:13]: Also here?

File: routes/graphqlVariablesExample.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+const { request, gql } = require('graphql-request')
+
```
Comment (EvgeniiaVak) [inline routes/graphqlVariablesExample.js:2]: @leomrocha We can probably safely delete this file?

File: README.md [hunk 4]
```diff
@@ -7,3 +7,70 @@ For data management systems including [CKAN](https://ckan.org/).
...
+- ~~offset (int) – query offset value~~ Not implemented
+- ~~total (int) – number of total matching records~~ Not implemented
+
+## Examples
+
+### Help page
+
+![Help Page](documentation/help-screen.png)
+
+### Basic query with limit
```
Comment (EvgeniiaVak) [inline README.md:58]: I think we need to have both URLs (to be able to easily copy-paste) and images here.

File: routes/index.js [hunk 5]
```diff
@@ -1,9 +1,113 @@
...
+  <ul>
+  <li>schema (JSON) – The data schema </li>
+  <li>data (JSON)  – matching results in JSON format</li>
+  </ul>
+  `
+  res.send(helpTxt)
+})
+
+/* GET . */
+router.get(`/${APP_VERSION}/datastore_search`, async function (req, res, next) {
```
Comment (EvgeniiaVak) [inline routes/index.js:81]: Maybe we should move this endpoint to its won router file?

File: routes/queryGraphQL.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
+const { request, gql } = require('graphql-request')
```
Comment (EvgeniiaVak) [inline routes/queryGraphQL.js:1]: And this file should probably be out of routes folder - just to make code reading easier in the future

File: test/integration/testQueryGraphQL.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    {
+      name: 'time_column',
+      type: { name: 'timestamp' },
+    },
+  ],
+}
+
+describe('graphqlQueryBuilder', function () {
+  // describe('function queryForData', function () {
```
Comment (EvgeniiaVak) [inline test/integration/testQueryGraphQL.js:34]: Are these outdated tests and should be removed?

**Review by EvgeniiaVak — APPROVED** [review]
Looks good 👍 We have some functionality 🎉  . I'll merge (and want to watch what will happen with the docker build) I also added some comments for code structure - @leomrocha please consider them.

File: routes/graphqlVariablesExample.js (hunk 3 above)
Comment (leomrocha) [reply routes/graphqlVariablesExample.js:2]: @EvgeniiaVak absolutely, we need to delete it

File: README.md (hunk 4 above)
Comment (leomrocha) [reply README.md:58]: I don't understand which urls you mean by both

File: routes/index.js (hunk 5 above)
Comment (leomrocha) [reply routes/index.js:81]: We could do it in a next PR

File: routes/queryGraphQL.js (hunk 6 above)
Comment (leomrocha) [reply routes/queryGraphQL.js:1]: Yes, we should refactor the structure here. Good catch

File: routes/graphqlVariablesExample.js (hunk 3 above)
Comment (leomrocha) [reply routes/graphqlVariablesExample.js:2]: Done

File: routes/queryGraphQL.js (hunk 6 above)
Comment (leomrocha) [reply routes/queryGraphQL.js:1]: Will change in a future refactoring.

File: test/integration/testQueryGraphQL.js (hunk 7 above)
Comment (leomrocha) [reply test/integration/testQueryGraphQL.js:34]: Done

File: ANALYSIS.md (hunk 1 above)
Comment (leomrocha) [reply ANALYSIS.md:5]: Fixed

**Pull request merged**